webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx

```diff
@@ -109,7 +119,10 @@ const ExperimentVisualization: React.FC<Props> = ({
         setValidationMetrics(newValidationMetrics);
         if (event.searcherMetric) setSearcherMetric(event.searcherMetric);
       },
-    );
+    ).catch(() => {
+      setHasLoaded(true);
```

Comment:
I think we could use a type like the one we defined in useRest for async request which has: `error` `hasLoaded` `data` or something similar here and many other places where we make api calls

Reply:
We might be primed for something like this for the new api / consumeStream!